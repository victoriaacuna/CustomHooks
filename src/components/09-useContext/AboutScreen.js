import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const AboutScreen = () => {

    const {user, setUser} = useContext(UserContext);

    return (
        <div>
            <h1>About Screen</h1>
            <hr/>

            <h3>{user?.name} {user?.last_name}</h3>
            <button className="btn btn-primary" onClick={ () => { setUser( {} ) } }>
                Logout
            </button>
        </div>
    )
}
