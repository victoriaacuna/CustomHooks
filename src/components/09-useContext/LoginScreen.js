import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {

    const { setUser} = useContext(UserContext);

    return (
        <div>
            <h1>Login Screen</h1>
            <hr/>

            <button className="btn btn-primary" onClick={ () => {setUser({
                id: 1234,
                name: 'Victoria',
                last_name: 'Acuna',
                email: 'vics@gmail.com'
            })}}>
                Cambiar user
            </button>
        </div>
    )
}
