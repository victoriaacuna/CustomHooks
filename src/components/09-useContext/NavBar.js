import React from 'react';
import {
    Link, NavLink
  } from "react-router-dom";

// La diferencia entre el NavLink y el Link es que el NavLink permite saber en quÃ© ruta me 
// encuentro y activar clases de css

export const NavBar = () => {
    return (
        <div>
            <ul>
                <li>
                    <Link to ="/" className="nav-item nav-link">
                        Home
                    </Link>
                </li>
                <li>
                    <NavLink exact activeClassName="active" className="nav-item nav-link" to ="/about">
                        About
                    </NavLink>
                </li>
                <li>
                    <Link to ="/login" className="nav-item nav-link">
                        Login
                    </Link>
                </li>
            </ul>
        </div>
    )
}
