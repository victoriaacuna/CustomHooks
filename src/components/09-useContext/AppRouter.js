import React from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect
  } from "react-router-dom";
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';
import { NavBar } from './NavBar';

export const AppRouter = () => {
    return (
        <Router>
            <>
                <NavBar/>

                <Switch>
                    <Route exact path="/" component={HomeScreen}/>
                    <Route exact path="/about" component={AboutScreen}/>
                    <Route exact path="/login" component={LoginScreen}/>

                    // Redirige si no encuentra ninguna ruta (por defecto).
                    <Redirect to="/"></Redirect>
                    
                    // Componente que muestra por defecto si la ruta no existe.
                    {/* <Route component={HomeScreen}/> */}

                </Switch>
            </>
        </Router>
    )
}
