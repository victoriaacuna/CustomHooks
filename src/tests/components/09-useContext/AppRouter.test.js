import '@testing-library/jest-dom';
import { mount } from 'enzyme';
import React from 'react';
import { AppRouter } from '../../../components/09-useContext/AppRouter';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Pruebas en el App Router', () => {

    const user = {
        id: 1,
        name: 'Victoria',
    }

    const wrapper = mount(
        <UserContext.Provider value = {{
            user
        }}>
            <AppRouter/>
        </UserContext.Provider>
    );

    test('should match snapshot', () => {

        expect(wrapper).toMatchSnapshot();

    });


})
