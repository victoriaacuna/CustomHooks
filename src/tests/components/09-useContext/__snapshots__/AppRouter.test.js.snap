// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en el App Router should match snapshot 1`] = `
Array [
  <div>
    <ul>
      <li>
        <a
          className="nav-item nav-link"
          href="/"
          onClick={[Function]}
        >
          Home
        </a>
      </li>
      <li>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/about"
          onClick={[Function]}
        >
          About
        </a>
      </li>
      <li>
        <a
          className="nav-item nav-link"
          href="/login"
          onClick={[Function]}
        >
          Login
        </a>
      </li>
    </ul>
  </div>,
  <div>
    <h1>
      Home Screen
    </h1>
    <hr />
    <pre>
      {
   "id": 1,
   "name": "Victoria"
}
    </pre>
  </div>,
]
`;
